### PR TITLE
Support text/csv response

### PR DESCRIPTION
**Pull request by duongvy0112**
issue: https://github.com/Unstructured-IO/community/issues/12
- Added new function `convert_to_isd_csv`
- Updated `pipeline_api`

### CODE REVIEW

**Comment by cragwolfe** [comment]
@duongvy0112 , couple of quick comments:

* need to run `make generate-api` to create the updated fastAPI route (https://github.com/Unstructured-IO/pipeline-sec-filings/blob/main/prepline_sec_filings/api/section.py ) from the notebook
* run `make tidy-notebooks` to clean metadata

Once that's done, it'd be good to have a unittest to test the updated FastAPI to retrieve text/csv.

**Comment by duongvy0112** [comment]
`make test` looks good

**Comment by cragwolfe** [comment]
Looks good, couple of minor things:

Please add a test to https://github.com/Unstructured-IO/pipeline-sec-filings/blob/main/test_sec_filings/sec_filings/test_section_api.py so there is test coverage for the newly added code.
Please add a single bullet to the CHANGELOG.md showing support for text/csv.

It would also be good to test this functionality with a real filing (like one of the filings that are downloaded into `sample-docs/` after running `make dl-test-artifacts`), to make sure non-ascii unicode characters do no pose a problem to the CSV writer.

File: pipeline-notebooks/pipeline-section.ipynb [hunk 1]
```diff
@@ -971,7 +1006,12 @@
     "        with timeout(seconds=5):\n",
     "            section_elements = sec_document.get_section_narrative(regex_enum)\n",
     "            results[f\"REGEX_{i}\"] = section_elements\n",
-    "    return {section:convert_to_isd(section_narrative) for section, section_narrative in results.items()}"
+    "    if response_type == \"application/json\":\n",
+    "        return {section:convert_to_isd(section_narrative) for section, section_narrative in results.items()}\n",
+    "    elif response_type == \"text/csv\":\n",
+    "        return convert_to_isd_csv(results)\n",
+    "    else:\n",
+    "        return \"unsupported response type\""
```
Comment (cragwolfe) [inline pipeline-notebooks/pipeline-section.ipynb:1014]: would be better to raise a `ValueError` here instead

**Comment by duongvy0112** [comment]
- raise a `ValueError`
- Added test function `test_section_narrative_api_with_text_csv_response_type`
- Test for writing csv file with the real filing (in sample-docs after `make dl-test-artifacts`)
- Updated `CHANGELOG.md`
- Updated `.gitignore` to skip .csv files

**Comment by duongvy0112** [comment]
`make check` failed because the test function name `def test_section_narrative_api_with_text_csv_response_type(form_type, response_type, section, tmpdir):` is longer than 100 (103 exactly)
Line 164 in Makefile: `black --line-length 100 test_${PIPELINE_PACKAGE} --check`
 The name should be updated?

**Comment by cragwolfe** [comment]
`make tidy` should clean that up

**Comment by duongvy0112** [comment]
The function `convert_to_isd_csv` in [pipeline-section.ipynb](https://github.com/Unstructured-IO/pipeline-sec-filings/blob/c6d5a0a23ad58e08b5311612c0ae14265ff016ba/pipeline-notebooks/pipeline-section.ipynb) could be moved to https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/staging/base.py , and replace it's function [convert_to_isd_csv](https://github.com/Unstructured-IO/unstructured/blob/6291d12f15342b9e966f14451e994c1f569490c1/unstructured/staging/base.py#L34) ?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+## 0.2.2-dev
+
+* Update `pipeline_api` to support `text/csv` response
+* Add new function `convert_to_isd_csv`
+* Add test function `test_section_narrative_api_with_text_csv_response_type`
```
Comment (cragwolfe) [inline CHANGELOG.md:5]: Apologies for not being more specific on the CHANGELOG update. Just one bullet mentioning CSV support is needed, added underneath `0.2.1-dev` (at line 9, pushing existing line down) is sufficient.

When it is just a `-dev update`, a new version is not needed.

Comment (duongvy0112) [reply CHANGELOG.md]: Just added `Support text/csv response type` underneath `0.2.1-dev`

**Review by cragwolfe — APPROVED** [review]

**Comment by cragwolfe** [comment]
Looks good, just need to `Update branch` against `main` so that a fully up to date branch passes CI.

**Pull request merged**